app/guides/[slug]/page.tsx

```diff
@@ -0,0 +1,136 @@
+
+const GuideTitle = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="guideTitle">{children}</div>;
+};
+
+const TableOfContents = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="pb-8 tableOfContents">{children}</div>;
+};
+
+const TwoColumn = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="grid grid-cols-1 lg:grid-cols-2 gap-8 pb-8">{children}</div>;
+};
+
+const ThreeColumn = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="grid grid-cols-1 lg:grid-cols-3 gap-6 pb-8">{children}</div>;
+};
+
+const Column = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="">{children}</div>;
+};
+
+const CenteredText = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="text-center">{children}</div>;
+};
+
+const CenteredContent = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="flex justify-center py-4">{children}</div>;
+};
+
+const Space = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="py-4" />;
+};
+
+const Thumbnail = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="max-w-16">{children}</div>;
+};
+
+const MiniImage = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="max-w-40">{children}</div>;
+};
+
+const ExtraSmallImage = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="max-w-xs">{children}</div>;
+};
+
+const SmallImage = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="max-w-md">{children}</div>;
+};
+
+const MediumImage = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="max-w-xl">{children}</div>;
+};
+
+const LargeImage = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="max-w-3xl">{children}</div>;
+};
+
+const ExtraLargeImage = ({ id, children }: { id: string; children: ReactNode }) => {
+  return <div className="max-w-5xl">{children}</div>;
+};
+
+// use markdown-components.ts if possible? Like this:
+// import { MarkdownComponents } from "@/lib/markdown-components";
+const Components = {
+  "guide-title": GuideTitle,
+  "toc": TableOfContents,
+  "column": Column,
+  "two-column": TwoColumn,
+  "three-column": ThreeColumn,
+  "centered-content": CenteredContent,
+  "centered-text": CenteredText,
+  "space": Space,
+  "thumbnail": Thumbnail,
+  "xxs-img": MiniImage,
+  "xs-img": ExtraSmallImage,
+  "sm-img": SmallImage,
+  "md-img": MediumImage,
+  "lg-img": LargeImage,
+  "xl-img": ExtraLargeImage,
+};
+
+export default async function Guides({ params }: { params: { slug: string } }) {
+  const guideContent = await getGuideContent(params.slug);
+  const guidePages = await getGuidePages();
+
+  function getSections(): any[] {
```

Comment:
i'm also a bit confused why this is needed. the NavMenu component should be the one that handles this logic once it gets the guides JSON data from the CMS in my opinion

Reply:
So the reason I did this here was because I didn't want to do special logic in the NavMenu code.
For building menus, I wanted it to be able to just take
```typescript
[
  { 
    name: string, 
    items: {
      title: string,
      href: string,
      description: string,
     }[] 
   }
]
```
If that makes sense. So we can use this elsewhere. I can do a helper function for this in lib and make it more "generic" sure.